operator/iamkeycontroller/observe.go

```diff
@@ -29,20 +31,34 @@ func (p *IAMKeyPipeline) Observe(ctx context.Context, mg resource.Managed) (mana
 		if KeyId, exists := iamKey.Annotations[KeyIDAnnotationKey]; exists {
 			iamKey.Status.AtProvider.KeyID = KeyId
 			delete(iamKey.Annotations, KeyIDAnnotationKey)
+			log.V(1).Info("Deleting annotation", "key", KeyIDAnnotationKey)
 		} else {
 			// New resource, create user first
+			log.V(1).Info("IAM key not found, returning")
 			return managed.ExternalObservation{}, nil
 		}
 	}
 
 	pctx := &pipelineContext{Context: ctx, iamKey: iamKey}
-	err := p.getIAMKey(pctx)
-	if err != nil {
-		return managed.ExternalObservation{}, resource.Ignore(isNotFound, err)
+
+	// since new create() adds RoleID annotation it's enought to just check whether it exists
+	if _, ok := iamKey.ObjectMeta.Annotations[RoleIDAnnotationKey]; ok {
```

Comment:
Shouldn't we delete this annotation and write it to the status? Like we do for `KeyIDAnnotationKey`.

See line 31 for how that annotation is handled.

Reply:
yeah, migrated into that logic